source/PlayerInfo.cpp

```diff
@@ -769,6 +769,61 @@ const shared_ptr<Ship> &PlayerInfo::FlagshipPtr()
 }
 
 
+// Move the flagship to the start of your list of ships. It does not make
+// sense that the flagship would change if you are reunited with a different
+// ship that was higher up the list. This function should not be used during
+// flight, since all shared-pointers to ships in the ships-list would be
+// invalidated (causing all kind of actions like ship and/or missile targeting
+// to be reset).
+void PlayerInfo::MoveFlagshipBegin()
+{
+	if(flagship)
+	{
+		auto it = find(ships.begin(), ships.end(), flagship);
+		if(it != ships.begin() && it != ships.end())
+		{
+			ships.erase(it);
+			ships.insert(ships.begin(), flagship);
+		}
+	}
+}
+
+
+
+// Set the flagship (on departure or during flight).
+void PlayerInfo::SetFlagship(shared_ptr<Ship> newFlagship)
```

Comment:
```suggestion
void PlayerInfo::SetFlagship(const shared_ptr<Ship> &newFlagship)
```

Reply:
As noted above; I prefer to use the shared_ptrs as pointers, not as references to pointers.